### ISSUE TITLE
Optimize gradient estimator given stochastic computation graph

### ISSUE THREAD

**Issue by HEmile** [HEmile/storchastic#2]
Implement the algorithm from https://arxiv.org/pdf/1506.05254.pdf. 

### Storch backward
`storch.backward` computes the derivative with respect to all `torch.Tensor`s with `requires_grad` by estimating the derivatives with respect to all registered cost nodes. To do this, it optimizes the graph and adds appropriate log probabilities (score functions) and baselines.
